File: backend/src/main/java/hu/bsstudio/raktr/dto/container/ContainerUpdateDto.java
package hu.bsstudio.raktr.dto.container;

import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ContainerUpdateDto {

    @NotBlank
    private String assetTag;

    @NotBlank
    private String barcode;

    @NotBlank
    private String name;

    @Min(1)
    @NotNull
    private Integer weight;

    private Boolean publicRentable;

    @NotBlank
    private String categoryName;

    @NotBlank
    private String locationName;

    @Min(1)
    @NotNull
    private Integer ownerId;

}

File: frontend/src/app/model/scannable/container/containerUpdateDto.ts
export class ContainerUpdateDto {
  assetTag: string;
  barcode: string;
  name: string;
  weight: number;
  publicRentable: boolean;
  categoryName: string;
  locationName: string;
  ownerId: number;

  constructor(
    assetTag: string,
    barcode: string,
    name: string,
    weight: number,
    publicRentable: boolean,
    categoryName: string,
    locationName: string,
    ownerId: number
  ) {
    this.assetTag = assetTag;
    this.barcode = barcode;
    this.name = name;
    this.weight = weight;
    this.publicRentable = publicRentable;
    this.categoryName = categoryName;
    this.locationName = locationName;
    this.ownerId = ownerId;
  }

  toJson(): Record<string, unknown> {
    return {
      assetTag: this.assetTag,
      barcode: this.barcode,
      name: this.name,
      weight: this.weight,
      publicRentable: this.publicRentable,
      categoryName: this.categoryName,
      locationName: this.locationName,
      ownerId: this.ownerId
    };
  }
}
